scripts/prune_lm_dir.py

```diff
@@ -425,17 +425,37 @@ def IterateOnce(threshold, step, iter, recovery_step, recovery_size):
                 break
 
     """
-    global logprob_changes, steps, current_num_ngrams
+    global logprob_changes, logprob_changes_for_KL, steps, current_num_ngrams
```

Comment:
Maybe clearer to write
 logprob_changes_without_dropped_steps
which better explains what the variable really represents.

On Thu, Jul 28, 2016 at 10:18 PM, Wang Jian <notifications@github.com>
wrote:

> In scripts/prune_lm_dir.py
> <https://github.com/danpovey/pocolm/pull/35#discussion_r72742640>:
>
> > @@ -425,17 +425,37 @@ def IterateOnce(threshold, step, iter, recovery_step, recovery_size):
> >                  break
> >
> >      """
> > -    global logprob_changes, steps, current_num_ngrams
> > +    global logprob_changes, logprob_changes_for_KL, steps, current_num_ngrams
>
> these values are used to compute the approximate K-L divergence with
> -sum(logprob_changes_for_KL)). In case that we go too far and recover from
> the recovery_step, i think we should overwrite the logprob_change for the
> dropped steps.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/35/files/e439a75744d11dc5ffa124298da6b8c461c0ce96#r72742640>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu4sm7Dwzn0xk69h17ZdN97HaXDYEks5qaY0SgaJpZM4JX4Rv>
> .
>